src/Lazy/pipeLazy.ts

```diff
@@ -0,0 +1,333 @@
+import pipe1 from "../pipe1";
+import reduce from "../reduce";
+import type Awaited from "../types/Awaited";
+import type ReturnPipeType from "../types/ReturnPipeType";
+
+/**
+ * Make function, that performs left to right function composition.
+ * All arguments must be unary.
+ *
+ * @example
+ * ```ts
+ * pipeLazy(
+ *  map(a => a + 10),
+ *  filter(a => a % 2 === 0),
+ *  toArray,
+ * )([1, 2, 3, 4, 5]); // [12, 14]
+ *
+ * await pipeLazy(
+ *  map(a => a + 10),
+ *  filter(a => a % 2 === 0),
+ *  toArray,
+ * )(Promise.resolve([1, 2, 3, 4, 5])); // [12, 14]
+ *
+ * // if you want to use asynchronous callback
+ * await pipeLazy(
+ *  toAsync,
+ *  map(async (a) => a + 10),
+ *  filter((a) => a % 2 === 0),
+ *  toArray,
+ * )(Promise.resolve([1, 2, 3, 4, 5])); // [12, 14]
+ *
+ * // with toAsync
+ * await pipeLazy(
+ *  toAsync,
+ *  map(a => a + 10),
+ *  filter(a => a % 2 === 0),
+ *  toArray,
+ * )([Promise.resolve(1), Promise.resolve(2), Promise.resolve(3), Promise.resolve(4), Promise.resolve(5)]); // [12, 14]
+ * ```
+ *
+ * {@link https://codesandbox.io/s/fxts-toarray-fy84i | Try It}
```

Comment:
@ppeeou @shine1594 can you please fix/correct only this part (codesandbox, doc, doc link)

Reply:
sure :)

Could you add type test? 
you can see reference https://github.com/marpple/FxTS/tree/main/type-check

Reply:
> Could you add type test? you can see reference https://github.com/marpple/FxTS/tree/main/type-check

✅

Reply:
@ppeeou if you have another notices, I can fix it too